internal/backends/postgresql/postgresql.go

```diff
@@ -41,17 +41,26 @@ type stats struct {
 // collectionsStats returns statistics about tables and indexes for the given collections.
 //
 // If the list of collections is empty, then stats filled with zero values is returned.
-func collectionsStats(ctx context.Context, p *pgxpool.Pool, dbName string, list []*metadata.Collection) (*stats, error) {
-	if len(list) == 0 {
-		return new(stats), nil
-	}
-
+func collectionsStats(ctx context.Context, p *pgxpool.Pool, dbName string, list []*metadata.Collection, val bool) (*stats, error) {
 	var err error
+	if len(list) == 0 && val {
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/3518
-	q := `ANALYZE`
-	if _, err = p.Exec(ctx, q); err != nil {
-		return nil, lazyerrors.Error(err)
+		q := `ANALYZE`
+		if _, err = p.Exec(ctx, q); err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		return new(stats), nil
+	}
+	if val {
+		var placeholder []string
+		for _, c := range list {
+			placeholder = append(placeholder, c.TableName)
```

Comment:
Placeholder is `$1`, `$2`, etc. That slices is not a slice of placeholders.

Also, we should add PostgreSQL schema names and escape identifiers properly using `pgx.Identifier`.

Reply:
Like so ?
```
if refresh {
		var identifiers []string
		var placeholder metadata.Placeholder
		placeholders := make([]string, len(list))
		args := []any{}
		for _, c := range list {
			tableName := pgx.Identifier{dbName, c.TableName}.Sanitize()
			identifiers = append(identifiers, tableName)
		}
		for i,identifier := range identifiers{
			placeholders[i] = placeholder.Next()
			args = append(args, identifier)
		}
		q := fmt.Sprintf(`ANALYZE %s`, strings.Join(placeholders, ", "))
		if _, err := p.Exec(ctx, q, args... ); err != nil {
			return nil, lazyerrors.Error(err)
		}
	}
```

Reply:
@AlekSi ??

Reply:
As for `pgx.Identifier` – writing and pushing a code would allow us to review on GitHub with per-line comments, etc. It is much harder to review code written in the comment itself

Reply:
Will push a commit with these changes ASAP. Thank you for resolving my doubt !